daemon/media_socket.c

```diff
@@ -1172,8 +1193,16 @@ static int stream_packet(struct stream_fd *sfd, str *s, const endpoint_t *fsin,
 
 	/* return values are: 0 = forward packet, -1 = error/dont forward,
 	 * 1 = forward and push update to redis */
-	if (rwf_in)
+	if (rwf_in) {
 		handler_ret = rwf_in(s, in_srtp);
+	}
+
+	// If recording pcap dumper is set, then we record the call.
+	if (recording_pdumper != NULL && call->record_call) {
+		stream_pcap_dump(recording_pdumper, stream, s);
+		call->recording->packet_num++;
```

Comment:
I believe you are correct here. Originally we were writing a pcap file per stream so this was safe as each stream had it's own packet counter and was locked. There is a read lock on the call's master lock at the beginning of this function, but that would be insufficient for this. I will get a fix for this going.